lib/kube/proxy/forwarder.go

```diff
@@ -1772,7 +1772,15 @@ func (f *Forwarder) portForward(authCtx *authContext, w http.ResponseWriter, req
 		return nil, trace.Wrap(err)
 	}
 
+	onForwardConnection := func(addr string, success bool) {
+		if !success {
+			f.log.Warn("Failed to establish connection to %q.", addr)
+		}
+		// TODO: Consider emit audit log events. Here we have the target pod port and an actual success value.
+	}
+
 	onPortForward := func(addr string, success bool) {
```

Comment:
for this one I would print the request as a API request. The interesting part comes from the portforward ports which we aren't logging here

Reply:
reverted to the original behavior, emitting only once.

Reply:
Updated.